### ISSUE TITLE
gaps module clean-up

### ISSUE THREAD

**Issue by h-2** [seqan/seqan3#59]
depends on #58 

  * [x] rename `gapped_alphabet<>` to `gapped<>` (it will be `gapped<dna4>` for example so it is clear¹)
  * [x] remove `set_gap` and `is_gap` since one can now just do `= gap::GAP` and `== gap::GAP` in constant time
  * [ ] make `gapped<>` inherit more stuff from `union_composition` via `using`
  * [ ] since it is then only a few lines long, move it to `alphabet/gap/gap.hpp`
  * [x] add gap submodule meta-header, either `alphabet/gap.hpp` or `alphabet/gap/all.hpp` depending or where the library is then
  * [x] streamline documentation, in particular add `\param` documentation to functions and `\ingroup` to everything
  * [x] deduplicate the test cases, add to the generic test cases
  * [x] add `noexcept` where adequate

if you feel that these issues are not related, feel free to make separate PRs!

**Comment by marehr** [comment]
@h-2 not sure what you mean by  
> make `gapped<>` inherit more stuff from `union_composition` via `using`

**Comment by h-2** [comment]
* [ ] gap shouldn't implement seqan3::alphabet_concept directly, because it inherits from union_composition (which does)
  * why is gap deactivated for the generic alphabet_test?

**Comment by h-2** [comment]
@marehr also the gap module and the composition module create lots of these warnings:
```
</home/mi/h4nn3s/devel/seqan3/include/seqan3/alphabet/composition/>:2: warning: Found unknown command `\group'
</home/mi/h4nn3s/devel/seqan3/include/seqan3/alphabet/gap/>:2: warning: Found unknown command `\group'
/home/mi/h4nn3s/devel/seqan3/include/seqan3/alphabet/composition/all.hpp:39: warning: Found unknown command `\group'
/home/mi/h4nn3s/devel/seqan3/include/seqan3/alphabet/gap/all.hpp:39: warning: Found unknown command `\group'
/home/mi/h4nn3s/devel/seqan3/include/seqan3/alphabet/composition/all.hpp:39: warning: Found unknown command `\group'
/home/mi/h4nn3s/devel/seqan3/include/seqan3/alphabet/gap/all.hpp:39: warning: Found unknown command `\group'
```

**Comment by marehr** [comment]
new ticket #97
